internal/http/services/owncloud/ocdav/put.go

```diff
@@ -204,25 +225,44 @@ func (s *svc) handlePut(w http.ResponseWriter, r *http.Request, ns string) {
 	}
 
 	dataServerURL := uRes.UploadEndpoint
-	// TODO(labkode): do a protocol switch
-	httpReq, err := rhttp.NewRequest(ctx, "PUT", dataServerURL, r.Body)
+
+	// create the tus client.
+	c := tus.DefaultConfig()
+	c.Resume = true
+	c.HttpClient = rhttp.GetHTTPClient(ctx)
+	c.Store, err = memorystore.NewMemoryStore()
 	if err != nil {
-		log.Error().Err(err).Msg("error creating http request")
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	httpReq.Header.Set("X-Reva-Transfer", uRes.Token)
 
-	httpClient := rhttp.GetHTTPClient(ctx)
-	httpRes, err := httpClient.Do(httpReq)
+	log.Debug().
+		Str("header", tokenpkg.TokenHeader).
+		Str("token", tokenpkg.ContextMustGetToken(ctx)).
+		Msg("adding token to header")
+	c.Header.Set(tokenpkg.TokenHeader, tokenpkg.ContextMustGetToken(ctx))
+
+	tusc, err := tus.NewClient(dataServerURL, c)
 	if err != nil {
-		log.Error().Err(err).Msg("error doing http request")
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	defer httpRes.Body.Close()
 
-	if httpRes.StatusCode != http.StatusOK {
+	metadata := map[string]string{
+		"filename": path.Base(fn),
+		"dir":      path.Dir(fn),
+		//"checksum": fmt.Sprintf("%s %s", storageprovider.GRPC2PKGXS(xsType).String(), xs),
+	}
+
+	upload := tus.NewUpload(r.Body, length, metadata, "")
```

Comment:
In the request we don't have enough information to create an unique fingerprint to have resumable uploads, that's why is empty. This is all about the PUT request.

Reply:
We use TUS here because the tus library will translate PUT to POST/PATHC so the data servers needs to only implement TUS and not PUTs.

Reply:
tracked in https://github.com/cs3org/reva/issues/699